src/main/dialogs/captureToolDialog.js

```diff
@@ -0,0 +1,476 @@
+                // 2. Full System Settings Capture
+                fs.appendFileSync(captureDirName + "/settingsCapture.json", JSON.stringify(data.settingsCapture, null, 4));
+
+                electron.dialog.showMessageBox({
+                    message: "Capture diagnostics successfully taken and are now stored in this directory:\n" + captureDirName
+                });
+            },
+            function (err) {
+                electron.dialog.showMessageBox({
+                    message: "There was a problem capturing the diagnostics:\n" + err
+                });
+                fluid.log("Error running capture diagnostics. ", err);
+            }
+        );
+    }
+};
+
+/*
+ * Generates and returns a set of diagnostics for the current local machine.
+ *
+ * Returns a promise containing these.
+ */
+gpii.app.captureTool.generateDiagnostics = function (diagnosticsCollector) {
+    return fluid.promise.fireTransformEvent(diagnosticsCollector.events.onCollectDiagnostics, {});
+};
+
+/**
+ * Fetches the list of locally installed solutions and send them to the renderer
+ * on the `channelNotifier`.
+ *
+ * @param {gpii.app.channelNotifier} channelNotifier - The channel notifier to the renderer process.
+ * @param {gpii.flowManager.local} flowManager - The local flow manager.
+ */
+gpii.app.captureTool.channelGetInstalledSolutions = function (channelNotifier, flowManager) {
+    flowManager.capture.getInstalledSolutionsForCurrentDevice().then(
+        function (data) {
+            channelNotifier.events.sendingInstalledSolutions.fire(data);
+            var runningSolutions = {};
+            fluid.each(data, function (solutionEntry, solutionId) {
+                try {
+                    fluid.log("ABOUT TO CHECK PROCESS FOR: ", solutionId);
+                    var isRunningState = gpii.processReporter.handleIsRunning(solutionEntry);
+                    fluid.log("RUNNING STATE: ", isRunningState);
+                    if (isRunningState) {
+                        runningSolutions[solutionId] = solutionEntry;
+                    }
+                } catch (err) {
+                    fluid.log("Exception trying to look up", solutionEntry, " Error: ", err);
+                }
+            });
+            channelNotifier.events.sendingRunningSolutions.fire(runningSolutions);
+        },
+        function (err) {
+            channelNotifier.events.sendingInstalledSolutions.fire({isError: true, message: err});
+        }
+    );
+};
+
+/**
+ * This performs the task of capturing solutions using the flow manager, one solution at a time.
+ * While the flow manager can handle capturing multiple solutions in one go, because there is a
+ * likelihood of unknown situations in the real world with solutions and settings handlers, we go
+ * one at a time such that if there is a failure or exception thrown in one solution, it only
```

Comment:
Note that as written, you are actually issuing all of the capture operations simultaneously and not sequencing them - the comment should reflect this.

Reply:
Updated this comment to hopefully reflect what is actually happening.